core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala

```diff
@@ -462,99 +479,85 @@ object SubtractFeeFromOutputsFinalizer {
   }
 }
 
-/** Assumes the input transaction has had no fee considerations
-  * and that all inputs are P2WPKH. Subtracts the estimated fee
-  * in equal portions from the outputs with the specified ScriptPubKeys
-  */
-case class P2WPKHSubtractFeesFromOutputsFinalizer(
-    feeRate: FeeUnit,
-    spks: Vector[ScriptPubKey])
-    extends RawTxFinalizer {
-
-  override def buildTx(txBuilderResult: RawTxBuilderResult)(implicit
-      ec: ExecutionContext): Future[Transaction] = {
-    val dummyTxWit = TransactionWitness(
-      Vector.fill(txBuilderResult.inputs.length)(
-        P2WPKHWitnessV0(ECPublicKey.freshPublicKey, DummyECDigitalSignature)))
-
-    val wtx = txBuilderResult.toWitnessTransaction(dummyTxWit)
-
-    val outputsAfterFee =
-      SubtractFeeFromOutputsFinalizer.subtractFees(wtx, feeRate, spks)
-
-    Future.successful(
-      txBuilderResult.toBaseTransaction.copy(outputs = outputsAfterFee))
-  }
-}
+case class DualFundingInput(
+    scriptSignature: ScriptSignature,
+    maxWitnessLen: Int)
 
-/** Finalizes a dual-funded transaction given the InputInfos
-  * from both parties and the non-change output ScriptPubKey.
+/** Finalizes a dual-funded transaction given the DualFundingInputs
+  * from both parties, their change spks and the non-change spk.
   *
   * This includes adding the future fee of spending transactions
   * to the non-change output as well as subtracting relevant fees
   * from the change outputs. This finalizer filters dust outputs.
-  *
-  * Note: spendingVBytes may be 0
   */
 case class DualFundingTxFinalizer(
-    spendingVBytes: Long,
-    feeRate: FeeUnit,
-    fundingSPK: ScriptPubKey,
-    inputInfos: Vector[InputInfo])
+    offerInputs: Vector[DualFundingInput],
+    offerPayoutSPK: ScriptPubKey,
+    offerChangeSPK: ScriptPubKey,
+    acceptInputs: Vector[DualFundingInput],
+    acceptPayoutSPK: ScriptPubKey,
+    acceptChangeSPK: ScriptPubKey,
+    feeRate: SatoshisPerVirtualByte,
+    fundingSPK: ScriptPubKey)
     extends RawTxFinalizer {
+  private val sharedFutureFee = (feeRate * 125).satoshis.toLong
```

Comment:
can this `125` be a constant, it is not clear what it is for

Reply:
Once I've updated the spec I'll be commenting with links to it